src/Security/Permission/ContributorVoter.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Security\Permission;
+
+use CultuurNet\UDB3\Contributor\ContributorRepository;
+use CultuurNet\UDB3\Model\ValueObject\Identity\UUID;
+use CultuurNet\UDB3\Model\ValueObject\Web\EmailAddresses;
+use CultuurNet\UDB3\Role\ValueObjects\Permission;
+use CultuurNet\UDB3\Security\UserEmailAddressRepository;
+use CultuurNet\UDB3\StringLiteral;
+
+final class ContributorVoter implements PermissionVoter
+{
+    private UserEmailAddressRepository $repository;
+
+    private ContributorRepository $contributorRepository;
+
+    public function __construct(
+        UserEmailAddressRepository $repository,
+        ContributorRepository $contributorRepository
+    ) {
+        $this->repository = $repository;
+        $this->contributorRepository = $contributorRepository;
+    }
+
+    public function isAllowed(Permission $permission, StringLiteral $itemId, StringLiteral $userId): bool
+    {
+        return $this->repository->getEmailForUserId($userId->toNative()) &&
+            in_array(
+                $this->repository->getEmailForUserId($userId->toNative()),
+                $this->getEmailAddressesForItem($itemId->toNative())->toArray()
+            );
```

Comment:
* I'd suggest to put the return value of `$this->repository->getEmailForUserId($userId->toNative())` in a variable so you don't need to call it twice
* I'd also suggest to not fetch all the email addresses but add some kind of `hasEmailAddress` on the contributor repository that does a query to check that a given item id has the given email address